### DISCUSSION TITLE
Idea: when Dark's Dark code can be in a package, it should be

### DISCUSSION THREAD

**Discussion by StachuDotNet** [darklang/dark#4385] [Feedback]
https://github.com/darklang/dark/discussions/4384#discussioncomment-3386280

I've been imagining that, long-term,:
- what can be written in Dark (rather than dark repo), should be written in Dark
- out of the stuff that can be written in Dark: if it can be written in a Package, it should
- one's editing experience comes with several 'default' packages (maybe this is per-canvas?)

Does that make sense to you?
I think this is a nice way to provide a composable experience, and will push us to have a good package infrastructure and ecosystem.

**Comment by pbiggar** [comment]
I agree with the first two, with emphasis on "long-term".

> one's editing experience comes with several 'default' packages (maybe this is per-canvas?)

This feels very specific, I'm not sure what this really implies.

**Reply by StachuDotNet** [reply]
for example:
- a 'common refactorings' package exists
- when you create a new Dark canvas, that package is 'present', such that you can immediately use the functionality availed by the package
